integrators/symplectic_runge_kutta_nyström_integrator.hpp

```diff
@@ -81,17 +81,31 @@ class SymplecticRungeKuttaNyströmIntegrator
   SymplecticRungeKuttaNyströmIntegrator(
       serialization::FixedStepSizeIntegrator::Kind const kind,
       FixedVector<double, stages_> const& a,
-                                        FixedVector<double, stages_> const& b);
+      FixedVector<double, stages_> const& b);
 
-  void Solve(IntegrationProblem<ODE> const& problem,
-             Time const& step) const override;
+  void Solve(Instant const& t_final,
+             not_null<IntegrationInstance*> const instance) const override;
+
+  not_null<std::unique_ptr<IntegrationInstance>> NewInstance(
+    IntegrationProblem<ODE> const& problem,
+    typename IntegrationInstance::AppendState<ODE> append_state,
+    Time const& step) const override;
 
   static int const order = order_;
   static bool const time_reversible = time_reversible_;
   static int const evaluations = evaluations_;
   static CompositionMethod const composition = composition_;
 
  private:
+  struct Instance : public IntegrationInstance {
+    Instance(IntegrationProblem<ODE> problem,
+             AppendState<ODE> append_state,
+             Time step);
+    IntegrationProblem<ODE> const problem;
+    AppendState<ODE> const append_state;
+    Time const step;
+  };
```

Comment:
Resumability; for the adaptive case, store things for seamless resumability at the last natural point (and we'll want to make reaching exactly t_final an option).